### PR TITLE
request input...

### LINKED ISSUE #3425: Document request argument binders

### PR DESCRIPTION
**Pull request by ladybug**
Closes #3425. Committing what I have for input and review feedback... Am I on the right track with this?

### CODE REVIEW

**Comment by jameskleeh** [comment]
In general on the right track. I'll leave some comments

File: src/main/docs/guide/httpServer/customBinding.adoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+== Argument binding
```
Comment (jameskleeh) [inline src/main/docs/guide/httpServer/customBinding.adoc]: Even though the argument binding API is generic, this section should only be framed around the HTTP server and request binders.

File: src/main/docs/guide/httpServer/customBinding.adoc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+An argument binding allows for the binding of an argument's value from a source.
+Micronaut follows a specific search path for the resolution of api:core.bind.ArgumentBinder[ArgumentBinder]s through the
+api:http.bind.binders.ArgumentBinderRegistry[ArgumentBinderRegistry].  The selection process starts with annotation binders
+and then searches "by type" binders.  Selection of a binder can also be affected by other factors such as the request media
+type.
+
+Unlike api:core.convert.TypeConverter[TypeConverter] instances, api:core.bind.ArgumentBinder[ArgumentBinder]s can potentially
+handle complex objects and typically work in conjunction with a api:core.convert.value.ConvertibleValues[ConvertibleValues] instance
```
Comment (jameskleeh) [inline src/main/docs/guide/httpServer/customBinding.adoc]: I'm not sure what you're trying to convey here. It seems like you're trying to say that its common to use the conversion service, however that is really only typical for annotation based binders. I would perhaps break this up into a subsection for annotated and typed binders and given examples of each.

File: src/main/docs/guide/httpServer/customBinding.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+An api:core.bind.ArgumentBinder[ArgumentBinder] can either be registered as a bean or by META-INF/services. In the case
```
Comment (jameskleeh) [inline src/main/docs/guide/httpServer/customBinding.adoc:11]: Is that true? Can you point me to where in the code that is happening?

File: src/main/docs/guide/httpServer/customBinding.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+of the latter, it will be globally available at all times, whilst the former is only present when a api:context.BeanContext[BeanContext] is initialized
+
+== Example: custom argument bindings
+
+. Key interfaces/sub-interfaces are:
+
+ * api:http.bind.binders.AnnotatedArgumentBinder[AnnotatedArgumentBinder]
+
+    - An api:core.bind.ArgumentBinder[ArgumentBinder] whose lookup is driven by an annotation definition
```
Comment (jameskleeh) [inline src/main/docs/guide/httpServer/customBinding.adoc]: I would remove this and only reference the `AnnotatedRequestArgumentBinder`

File: src/main/docs/guide/httpServer/customBinding.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+ * api:http.bind.binders.AnnotatedRequestArgumentBinder[AnnotatedRequestArgumentBinder]
+
+    - An interface for classes that bind an api:core.type.Argument[Argument] from an api:http.HttpRequest[HttpRequest] driven by an annotation.
+
+ * api:http.bind.binders.TypeArgumentBinder[TypeArgumentBinder]
+
+    - an argument binder that is based on the argument type
```
Comment (jameskleeh) [inline src/main/docs/guide/httpServer/customBinding.adoc]: I would remove this and only reference the TypedRequestArgumentBinder

File: src/main/docs/guide/httpServer/customBinding.adoc (hunk 2 above)
Comment (ladybug) [reply src/main/docs/guide/httpServer/customBinding.adoc:11]: @jameskleeh I got this from the documentation of the interface ArgumentBinder

File: src/main/docs/guide/httpServer/customBinding.adoc (hunk 3 above)
Comment (ladybug) [reply src/main/docs/guide/httpServer/customBinding.adoc:11]: @jameskleeh I got this from the documentation of the interface ArgumentBinder

**Pull request closed without merging**